app/APIStart.py

```diff
@@ -8,148 +8,206 @@
 from flask import Flask
 from flask import request
 from flask import jsonify
-import json
-from DBhandler import DatabaseHandler as db
-from Items import Item as item
-from User import User
+from app.User import User, UserBase
+from app.Item import Item, ItemBase
+from sqlalchemy import create_engine
+from flask_sqlalchemy import SQLAlchemy
 
 app = Flask(__name__)
+database = 'lost_and_found'
+user = 'root'
+password = 'xzcv'
+host = 'localhost'
+port = '3306'
 
+"""-----------------------------------------------------------------------------"""
+"""This Section is to create database if it doesn't already exists"""
+connection_string = 'mysql://{0}:{1}@{2}:{3}'.format(user, password, host, port)
+# This engine just used to query for list of databases
+mysql_engine = create_engine(connection_string)
 
+# Query for existing databases
+existing_databases = mysql_engine.execute("SHOW DATABASES;")
+# Results are a list of single item tuples, so unpack each tuple
+existing_databases = [d[0] for d in existing_databases]
 
+# Create database if not exists
+if database not in existing_databases:
+    mysql_engine.execute("CREATE DATABASE {0}".format(database))
+"""--------------------------------------------------------------------------------"""
 
-@app.route('/')
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app.config['SQLALCHEMY_DATABASE_URI'] = connection_string + '/{}'.format(database)
+
+db = SQLAlchemy(app)
+
+
+@app.before_first_request
+def setup():
+    """Check Database if it exists with the Tables of our model"""
+    mysql_engine.execute("use {};".format(database))
+    existing_tables = mysql_engine.execute("Show Tables;")
+    existing_tables = [d[0] for d in existing_tables]
+    table = User.__tablename__
+    if table not in existing_tables:
+        UserBase.metadata.drop_all(bind=db.engine)
+        ItemBase.metadata.drop_all(bind=db.engine)
+        UserBase.metadata.create_all(bind=db.engine)
+        ItemBase.metadata.create_all(bind=db.engine)
+
+
+@app.route("/")
 def index():
-    return jsonify('This is index page')
+    return "Hello World"
+
 
-@app.route('/items/create',methods=['POST'])
+@app.route('/items/create', methods=['POST'])
 def create():
     try:
-        testdb = db()
         name = request.json['name']
         loc = request.json['location']
         desc = request.json['description']
-        items = item()
-        items.newItem(name,loc,desc)
-        testdb.insert_item_db(items)
-        
-        resp = jsonify({"Action":'Item Added Successfully'})
+        item = Item(name, loc, desc)
+        db.session.add(item)
+        db.session.commit()
+        db.session.close()
```

Comment:
what if we keep the session open?

Reply:
I did it because it is good practice to close or release resources that the programmer is using but I didn't know at first but upon searching "Why Session is closed in SQLAlchemy ?", What i found is that when committing a transaction some exception might occur while committing but in this case of code session.close() is not useful for further detail see the image attached
![Opera Snapshot_2019-11-13_110155_groups google com](https://user-images.githubusercontent.com/28764891/68737379-1418e380-0605-11ea-9ac5-c4e28853d2a3.png)
  In this scenario if an exception occurs while committing  it will be excepted